src/runtime/linux/auxv.rs

```diff
@@ -0,0 +1,142 @@
+//! Reads ELF Auxiliary Vector
+
+/// Reads the ELF Auxiliary Vector using libc's `getauxval`.
+pub use coresimd::__vendor_runtime::__runtime::linux::auxv::libc;
+
+/// Reads the ELF Auxiliary Vector from `/proc/self/auxv`.
+pub mod proc_self {
+    pub use coresimd::__vendor_runtime::__runtime::linux::auxv::{AT_HWCAP2,
+                                                                 AuxVec,
+                                                                 AT_HWCAP};
+    use std::mem;
+
+    /// Tries to read the ELF Auxiliary Vector from `/proc/self/auxv`.
+    ///
+    /// Errors if the file cannot be read. If a component of the auxvector
+    /// cannot be read, all the bits in its bitset are set to zero.
+    pub fn auxv() -> Result<AuxVec, ()> {
+        auxv_from_file("/proc/self/auxv")
+    }
+
+    fn auxv_from_file(file: &str) -> Result<AuxVec, ()> {
+        use std::io::Read;
+        let mut file = ::std::fs::File::open(file).or_else(|_| Err(()))?;
+
+        // See https://github.com/torvalds/linux/blob/v3.19/include/uapi/linux/auxvec.h
+        //
+        // The auxiliary vector contains at most 32 (key,value) fields: from
+        // `AT_EXECFN = 31` to `AT_NULL = 0`. That is, a buffer of
+        // 2*32 `usize` elements is enough to read the whole vector.
+        let mut buf = [0usize; 64];
+        {
+            let raw: &mut [u8; 64 * mem::size_of::<usize>()] =
+                unsafe { mem::transmute(&mut buf) };
+            file.read(raw).or_else(|_| Err(()))?;
+        }
+        auxv_from_buf(&buf)
+    }
+
+    fn auxv_from_buf(buf: &[usize; 64]) -> Result<AuxVec, ()> {
```

Comment:
I've refactor `auxv` into these three functions for testing purposes (see below). It allows testing the `auxv` functionality by reading a file, using `include_bytes`, or just adding some `auxv` arrays to the test module.